evap/staff/tests/test_importers.py

```diff
@@ -374,6 +387,117 @@ def test_replace_consecutive_and_trailing_spaces(self):
         success_messages, __, __ = EnrollmentImporter.process(excel_content, self.semester, None, None, test_run=True)
         self.assertIn("The import run will create 1 courses/evaluations and 3 users", "".join(success_messages))
 
+    def test_existing_course_is_not_recreated(self):
+        self.create_existing_course()
+        excel_content = excel_data.create_memory_excel_file(excel_data.test_enrollment_data_existing_course)
+
+        self.assertEqual(
+            len(
+                Course.objects.filter(
+                    semester=self.semester, name_de=self.existing_course.name_de, name_en=self.existing_course.name_en
+                )
+            ),
+            1,
+        )
+        __, warnings, __ = EnrollmentImporter.process(excel_content, self.semester, None, None, test_run=False)
+        self.assertIn(
+            "Course Existing Course (Existierender Kurs) already exists with identical attributes. Course is not created and users are put into the evaluation of that course.",
+            "".join(warnings[ImporterWarning.DUPL]),
+        )
+        self.assertEqual(
+            len(
+                Course.objects.filter(
+                    semester=self.semester, name_de=self.existing_course.name_de, name_en=self.existing_course.name_en
+                )
+            ),
+            1,
+        )
+
+    def test_existing_course_degree_is_added(self):
+        self.create_existing_course()
+        excel_content = excel_data.create_memory_excel_file(excel_data.test_enrollment_data_existing_course)
+
+        self.existing_course.degrees.set([Degree.objects.get(name_de="Master")])
```

Comment:
Agree -- this currently tests whether "Bachelor" will be added to the already existing "Master", we could just do it the other way around. (Or did you maybe want to test importing with both, "Bachelor" and "Master" already set?)

Reply:
What I did there was just me being lazy to work around creating another memory_excel_file data that would have the Master degree in it so it can be added to the existing course. Would you like to see another excel file data or rather a second variant of the `existing_course` instance?

Reply:
I'm not sure if I miss something...
If we remove the last line here (420) (which was the suggested change, from what I understood), the `memory_excel_file` would only add a Bachelor degree to `self.existing_course`, which already has a Bachelor degree. So we could not really assert that some degree got added.
Should I change the degree of self.existing_course to Master inside the `create_existing_course` function?
I think I dont know where the Master degree should come from, if line 420 is removed.

Reply:
Sorry, somehow, I was under the impression that the memory_excel_file contained both degrees for the course. I probably looked at the wrong line. Ignore my comments.

Reply:
I'm not sure what to do about this 😅...
Is it really a problem to rely on `degrees.set(...)`? I mean yeah, it could break... But every other component in django could break as well, or is there something special or unstable about `degrees.set(...)`?

Reply:
Yep seems valid. In `test_existing_course_different_attributes`, `test_existing_course_equal_except_evaluations` and `test_existing_course_different_grading` such adjustments are done as well. Do you think one one comment in `test_existing_course_degree_is_added` is enough or should we at least add references to it in the other tests?

Reply:
(for me, all methods, including this one, would be fine without the comment)